prow/cmd/deck/main.go

```diff
@@ -277,7 +279,7 @@ func main() {
 
 	// setup config agent, pod log clients etc.
 	configAgent := &config.Agent{}
-	if err := configAgent.Start(o.configPath, o.jobConfigPath); err != nil {
+	if err := configAgent.Start(o.configPath, o.jobConfigPath, spglassConfigDefaulting); err != nil {
```

Comment:
The reason for adding passing this in rather than defining it in the `config` package is that:
* This needs the `lens` package because it accesses its registr
* The lens package needs the `config` package, so importing the former into the later causes an import cycle

Even if the import cycle issue was solved, defining this in the `config` package would mean that we have to import all lenses everywhere where we use the `config` package (so more or less everywhere) and I think we should try to avoid that.

Reply:
Why should we avoid it? We do it for all other config for every part of the system. If we want to break from that let's do it in a focused effort?

Reply:
But in this particular case it implies that we do not only have to import some api types, we have to import everything because we need a populated registry. This is like adding a `_ "k8s.io/test-infra/prow/tide` import to plank.

Reply:
(Except that its not just for plank but for everything that uses the `config` package)